orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -208,6 +208,10 @@ def backward(ctx, *grad_outputs):
                     # Push user output grads to ONNX backend.
                     contiguous_grad_outputs = []
                     for idx, grad_output in enumerate(grad_outputs):
+                        if idx in self._onnx_graphs_info.output_grad_indices_non_differentiable:
+                            assert grad_output is None
```

Comment:
Coudl you provide a meaningful error message here? This is a user-facing error, so a hint on the issue might help them to have insights on the root cause

Reply:
done.